src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -158,7 +166,19 @@ class AuthScreens extends React.Component {
 
         // Fetch some data we need on initialization
         NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.NVP_PRIORITY_MODE, 'default');
-        NameValuePair.get(CONST.NVP.PREFERRED_LOCALE, ONYXKEYS.NVP_PREFERRED_LOCALE, 'en');
+
+        API.Get({
+            returnValueList: 'nameValuePairs',
+            nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        }).then((response) => {
+            const preferredLocale = response.nameValuePairs.preferredLocale || CONST.DEFAULT_LOCALE;
+            if (currentPreferredLocale !== CONST.DEFAULT_LOCALE && preferredLocale !== currentPreferredLocale) {
```

Comment:
would you mind adding parentheses to this logic to make it more clear. I know the order of operations, but I think we can increase readability